core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -741,7 +741,7 @@ public boolean append(Object kbase, long koff, int klen, Object vbase, long voff
         longArray.set(pos * 2 + 1, keyHashcode);
         isDefined = true;
 
-        if (numKeys >= growthThreshold && longArray.size() < MAX_CAPACITY) {
+        if (numKeys >= growthThreshold && longArray.size() / 2 < MAX_CAPACITY) {
```

Comment:
Probably, we need to leave some comments about why the size needs to be divided by 2.

Reply:
thanks! Added few comments.

Reply:
Thanks! It looks good to me.